t/plugin/proxy-rewrite.t

```diff
@@ -403,3 +403,223 @@ host: localhost
 x-real-ip: 127.0.0.1
 --- no_error_log
 [error]
+
+
+
+=== TEST 14: set route(rewrite headers)
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "plugins": {
+                            "proxy-rewrite": {
+                                "uri": "/uri/plugin_proxy_rewrite",
+                                "headers": {
+                                    "X-Api-Version": "v2"
+                                }
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/hello"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 15: rewrite headers
+--- request
+GET /hello HTTP/1.1
+--- more_headers
+X-Api-Version:v1
+--- response_body
+uri: /uri/plugin_proxy_rewrite
+host: localhost
+x-api-version: v2
```

Comment:
why not use `response-header` to judge headers?

Reply:
we used a fake server to response the request header for checking @moonming